modules/virt-creating-a-service-backed-by-virtual-machines.adoc

```diff
@@ -0,0 +1,142 @@
+// Module included in the following assemblies:
+//
+// * virt/virtual_machines/vm_networking/virt-using-the-default-pod-network-with-virt.adoc
+
+[id="virt-creating-a-service-backed-by-virtual-machines_{context}"]
+
+= Creating a service backed by virtual machines
+
+To create a service backed by a virtual machine, you start the `VirtualMachineInstance`,
+if it is not already running. Once the instance is running, you connect to the instance, and
+create a service object to expose the instance.
+
+The following service types are supported:
+
+* `ClusterIP` - Connect using service IP and service port inside the cluster network.
+* `NodePort` - Connect using node IP and node port outside the cluster network.
+* `LoadBalancer` - Connect using public IP and port.
+
+This procedure presents an example of how to create and expose a service object of type `ClusterIP`
+as a virtual machine-backed service.
+
+[NOTE]
+====
+Labels on a `VirtualMachineInstance` are passed through to the pod. The labels on
+the `VirtualMachineInstance` must match the labels on the Service object.
+====
+
+.Procedure
+
+. Edit the `VirtualMachineInstance` YAML configuration file to include the
+label `special: key:`.
++
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha3
+kind: VirtualMachineInstance
+metadata:
+  name: vmi-ephemeral
+  labels:
+    special: key <1>
+spec:
+  domain:
+    devices:
+      disks:
+      - disk:
+          bus: virtio
+        name: containerdisk
+    resources:
+      requests:
+        memory: 64M
+  volumes:
+  - name: containerdisk
+    containerDisk:
+      image: kubevirt/cirros-registry-disk-demo:latest
+----
+<1> The `special: key:` label
++
+
+. Edit the YAML to create and expose the Service object.
+
+.. Add `type: ClusterIP` for a
+ClusterIP service. Add `targetPort: 22`, exposing the service on SSH port `22`.
```

Comment:
Not sure we want to mention somewhere `ClusterIP` service type is the default.

Reply:
Mentioning defaults is always good. Added this NOTE:

NOTE 
====
ClusterIP is the default service type, if the service type is not specified.
====